flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -36,52 +35,49 @@
 
 import static org.apache.flink.runtime.state.ChannelPersistenceITCase.getStreamFactoryFactory;
 import static org.apache.flink.util.CloseableIterator.ofElements;
-import static org.apache.flink.util.ExceptionUtils.findThrowable;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** {@link ChannelStateWriterImpl} lifecycle tests. */
-public class ChannelStateWriterImplTest {
+class ChannelStateWriterImplTest {
     private static final long CHECKPOINT_ID = 42L;
     private static final String TASK_NAME = "test";
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testAddEventBuffer() throws Exception {
+    @Test
+    void testAddEventBuffer() throws Exception {
 
         NetworkBuffer dataBuf = getBuffer();
         NetworkBuffer eventBuf = getBuffer();
         eventBuf.setDataType(Buffer.DataType.EVENT_BUFFER);
-        try {
-            runWithSyncWorker(
-                    writer -> {
-                        callStart(writer);
-                        writer.addInputData(
-                                CHECKPOINT_ID,
-                                new InputChannelInfo(1, 1),
-                                1,
-                                ofElements(Buffer::recycleBuffer, eventBuf, dataBuf));
-                    });
-        } finally {
-            assertTrue(dataBuf.isRecycled());
-        }
+
+        runWithSyncWorker(
```

Comment:
I also noticed the following: Refactoring this code in a way that we're using `assertThatThrownBy` is used changes the behavior of `runWithSyncWorker` slightly because of `worker.processAllRequests()` in [ChannelStateWriterImplTest:313](https://github.com/apache/flink/blob/772de6fc8698eea8b7b42c3a1de4282a61ba7e8f/flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java#L313). The old implementation didn't call `processAllRequests()` if an exception was thrown. The refactored version would actually cause this method to be called which might change the behavior. You might want to double-check whether that's something we want.

Reply:
I checked all caller. All the unit tests of `@Test(expected = Exception.class)` throw an exception in the last line of runWithSyncWorker or it just has one line, so I think the old and new codes are the same.

Reply:
I guess, you misunderstood me here: `worker.processAllRequests()` in [ChannelStateWriterImplTest:319](https://github.com/apache/flink/blob/5596204e67027ff924b97616b5793c8073434412/flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java#L319) is called after the callback function is executed. In the old code, we threw an Exception within the callback function which resulted in the command in line 319 not being called. In the new version of the code, we rely on `assertThatThrownBy` which doesn't throw an Exception and, therefore, line 319 is executed, i.e. `worker.processAllRequests()` is called in the expected case. This feels like an unwanted change? :thinking:

Reply:
When the callback doesn't throw exception. Old and new code is same.

Some old code tests rely on `runWithSyncWorker -> worker.processAllRequests()` to throw exceptions for tests that expect Exception. 

In the new code, I have additionally called worker.processAllRequests() in the callback, so `runWithSyncWorker -> worker.processAllRequests()` will not process any requests.

Please correct me, if anything is wrong. 

Or if we worry about which tests are wrong, we can refactor them into non-specific checks, what do you think?

Reply:
To be on the safe side, we should remove `worker.processAllRequests()` in [ChannelStateWriterImplTest:316](https://github.com/apache/flink/blob/3156ff6219452c51c010ded37197d01c1c5924c1/flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java#L316) and move it into the tests where this call is still required. This is test-specific code (for tests where no exception is expected) rather than generic code. WDYT?